### PR TITLE
openthread: Update to the latest upstream/master.

### PR DESCRIPTION
**Pull request by turon**
Similar to #12169.  This PR updates openthread to the latest upstream/master.
Both echo_server and echo_client compile and are operational.

### CODE REVIEW

File: subsys/net/l2/openthread/openthread.c [hunk 1]
```diff
@@ -235,10 +241,13 @@ int openthread_send(struct net_if *iface, struct net_pkt *pkt)
 	int len = net_pkt_get_len(pkt);
 	struct net_buf *frag;
 	otMessage *message;
+	otMessageSettings settings;
 
 	NET_DBG("Sending Ip6 packet to ot stack");
 
-	message = otIp6NewMessage(ot_context->instance, true);
+	settings.mLinkSecurityEnabled = true;
+	settings.mPriority            = OT_MESSAGE_PRIORITY_NORMAL;
```
Comment (rlubos) [inline subsys/net/l2/openthread/openthread.c]: No need to align assignment here.

File: subsys/net/lib/openthread/CMakeLists.txt [hunk 2]
```diff
@@ -23,7 +23,8 @@ if(NOT EXTERNAL_PROJECT_PATH_OPENTHREAD)
   # TODO: Point to a Zephyr fork
   # Nov. 7
   set_ifndef(ot_GIT_REPOSITORY "https://github.com/openthread/openthread.git")
-  set_ifndef(ot_GIT_TAG db4759cc41257d0572ddedbeead1e02c52033620)
+  set_ifndef(ot_GIT_TAG 2a75d30684654e0348bce9327c6f45f4e9ea71a5)
+  #set_ifndef(ot_GIT_TAG db4759cc41257d0572ddedbeead1e02c52033620)
```
Comment (rlubos) [inline subsys/net/lib/openthread/CMakeLists.txt]: Please remove the old entry instead of commenting it out.

**Review by rlubos — CHANGES_REQUESTED** [review]
Thank you for the update, really appreciate your work here.

Apparently, it turns out that new OpenThread is more hungry for stack, I've had to set `CONFIG_NET_RX_STACK_SIZE=5120` in `overlay-ot.conf` files in order to make the echo samples work as expected. If you have time, please update those values in `samples/net/echo_*` and `samples/net/sockets/echo_*` samples.

**Comment by rlubos** [comment]
And as for the commits themselves, we typically squash changes like in `ot: white space` into the original commit. Also different commit prefixing pattern is required by CI, preferably `net: openthread:` in this case. Finally, commit have to be signed and commit body is required. You could place OT commit used in the update in there.

**Comment by turon** [comment]
@rlubos, thanks for your comments and consideration.
I believe I've resolved everything you've mentioned.
Please have a look and let me know if I've missed anything.

**Review by rlubos — APPROVED** [review]
Thank you Martin for the update, works great!

**Review by carlescufi — APPROVED** [review]

**Pull request merged**